my-app/src/components/PostsList/Post/styled.ts

```diff
@@ -1,10 +1,33 @@
 import styled from 'styled-components';
 
-const StyledPost = styled.div<{ id: number}>`
+const StyledPost = styled.div<{ id: number, posts: {id: number}[], theme: 'light' | 'dark' }>`
     position: relative;
     padding-bottom: 71px;
-    border-bottom: 1px solid gainsboro; 
-
+    border-bottom: 1px solid gainsboro;
+    grid-area: ${({id, posts}) => {
```

Comment:
Using so many "if" block isn't really the best way to do this. Try this instead:

```
grid-area: ${({ id, posts }) => {
        const index = posts.findIndex(post => post.id === id);

        if (index !== -1) {
            const row = Math.floor(index / 2) * 2 + 1; // Calculate the row position
            return `${row} / 1 / ${row + 2} / 2`;
        }

        return ''; // Default grid area
    }};
```